src/python/TaskWorker/Worker.py

```diff
@@ -36,13 +36,17 @@ def addTaskLogHandler(logger, username, taskname):
     #set the logger to save the tasklog
     formatter = logging.Formatter("%(asctime)s:%(levelname)s:%(module)s:%(message)s")
     taskdirname = "logs/tasks/%s/" % username
-    if not os.path.isdir(taskdirname):
+    try:
         os.mkdir(taskdirname)
+    except OSError as ose:
+        if ose.errno != 17: #ignore the "Directory already exists error" but print other errors traces
+            logger.exception("Cannot set task hanlder logfile for task %s. Ignoring and continuing normally." % taskname)
```

Comment:
if you meant"
  "logger calls to log/task/taskname.log will silently fail and all messages will be in twlog.txt only",
that's all I wanted to know. I agree with all other things you just wrote, and was not questioning 
them !